cider/function/datetime/ExtractFromTime.cpp

```diff
@@ -161,12 +165,20 @@ extern "C" RUNTIME_EXPORT ALWAYS_INLINE int64_t extract_dateepoch(const int64_t
 }
 
 // First day of epoch is Thursday, so + 3 to have Monday=0, then + 1 at the end.
-extern "C" RUNTIME_EXPORT ALWAYS_INLINE int64_t extract_isodow(const int64_t timeval) {
+extern "C" RUNTIME_EXPORT ALWAYS_INLINE int64_t
+extract_isodow(const int64_t timeval, const bool is_null = false) {
+  if (is_null == true) {
+    return 0;
+  }
   int64_t const days_past_epoch = floor_div(timeval, kSecsPerDay);
   return unsigned_mod(days_past_epoch + 3, kDaysPerWeek) + 1;
 }
 
-extern "C" RUNTIME_EXPORT ALWAYS_INLINE int64_t extract_day(const int64_t timeval) {
+extern "C" RUNTIME_EXPORT ALWAYS_INLINE int64_t extract_day(const int64_t timeval,
+                                                            const bool is_null = false) {
+  if (is_null == true) {
```

Comment:
how about using macro to generate both `extract_day`/`extract_day_nullable` methods?